src/clean-core/alloc_array.hh

```diff
@@ -86,22 +85,25 @@ struct alloc_array
     alloc_array(alloc_array const& a) = delete;
     alloc_array& operator=(alloc_array const& a) = delete;
 
-    ~alloc_array()
-    {
-        detail::container_destroy_reverse<T>(_data, _size);
-        _free(_data);
-    }
+    ~alloc_array() { _destroy(); }
 
     void resize(size_t new_size, T const& value = {})
     {
-        detail::container_destroy_reverse<T>(_data, _size);
-        _free(_data);
+        _destroy();
         _size = new_size;
         _data = _alloc(new_size);
         for (size_t i = 0; i < new_size; ++i)
             new (placement_new, &_data[i]) T(value);
     }
 
+    void destroy()
```

Comment:
should this really be public API? we don't have this in any other container. an array can be reset via ` = {};` as well